tests/test_sound.py

```diff
@@ -0,0 +1,379 @@
+"""Tests for generating sound stimuli.
+
+This script runs tests that generate different sound stimuli and verifies
+that they are initialized correctly.
+
+Currently these only work if AUDIOSERVER is 'jack'. 'pyo' is not tested.
+'docs' doesn't actually generate waveforms.
+
+This doesn't require (or test) a running jackd or even a JackClient.
+Instead, these tests short-circuit those dependencies by manually setting
+FS and BLOCKSIZE in autopilot.stim.sound.jackclient.
+
+A TODO is to test the JackClient itself.
+
+Currently only the sound Noise is tested.
+
+These tests cover multiple durations and amplitudes of mono and
+multi-channel Noise, including some edges cases like very short durations
+or zero amplitude.
+
+The rest of this docstring addresses the workaround used to short-circuit
+jackd and JackClient.
+
+Here is the sequence of events that leads to FS and BLOCKSIZE.
+* If an autopilot.core.pilot.Pilot is initialized:
+**  autopilot.core.pilot.Pilot.__init__ checks prefs.AUDIOSERVER,
+    and calls autopilot.core.pilot.Pilot.init_audio.
+**  autopilot.core.pilot.Pilot.init_audio calls 
+    autopilot.external.__init__.start_jackd.
+**  autopilot.external.__init__.start_jackd takes the JACKDSTRING pref 
+    and replaces the token '-rfs' in it with the FS pref. The jackd
+    process is launched and stored in autopilot.external.JACKD_PROCESS.
+    That process may fail or not, we continue anyway.
+**  Next, autopilot.core.pilot.Pilot.init_audio instantiates an
+    autopilot.stim.sound.jackclient.JackClient()
+**  autopilot.stim.sound.jackclient.JackClient.__init__ 
+    initalizes a jack.Client
+**  autopilot.stim.sound.jackclient.JackClient.fs 
+    is set to jack.Client.samplerate. Note that this is either the
+    requested sample rate, or some default value from jack (not Autopilot)
+    if the client did not actually succeed in booting.
```

Comment:
yes we should have some check if the fs doesn't match that requested in prefs

Reply:
done in https://github.com/wehr-lab/autopilot/pull/102/commits/4fe7990b4f19a59b0a9d52cfccbc95d2db825784

Reply:
encountered these kinds of issues when writing and running these tests on my desktop PC, instead of on an rpi ... kind of a good way to see what depends on jack and an expected audio device, and what doesn't